week2/exercises/02-schema.md

```diff
@@ -0,0 +1,30 @@
+# Schema
+
+Create a new database/schema `hyf_node_week2`, then create the following tables:
+
+```sql
+CREATE TABLE `users` (
+  `id` int unsigned NOT NULL AUTO_INCREMENT,
+  `created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `confirmed_at` datetime DEFAULT NULL,
+  `first_name` varchar(255) NOT NULL,
+  `last_name` varchar(255) NOT NULL,
+  `email` varchar(255) NOT NULL,
+  `token` varchar(255) DEFAULT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `email_UNIQUE` (`email`),
+  UNIQUE KEY `token_UNIQUE` (`token`)
+) ENGINE=InnoDB AUTO_INCREMENT=11 DEFAULT CHARSET=utf8mb4;
+
+CREATE TABLE `snippets` (
+  `id` int unsigned NOT NULL AUTO_INCREMENT,
+  `created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `user_id` int unsigned NOT NULL,
+  `title` varchar(255) NOT NULL,
+  `contents` text NOT NULL,
+  `is_private` tinyint NOT NULL DEFAULT '1',
+  PRIMARY KEY (`id`),
+  KEY `id_idx` (`user_id`),
+  CONSTRAINT `id` FOREIGN KEY (`user_id`) REFERENCES `users` (`id`) ON DELETE RESTRICT
```

Comment:
Not CASCADE?

Reply:
My thought was to prevent accidentally deleting a user with snippets. To resolve that they would have to delete their snippets first.